src/routes/session.ts

```diff
@@ -5,26 +5,71 @@ import {
 	comparePassword,
 	runInvalidPasswordComparison,
 } from '../platform/auth';
-import { sbvrUtils } from '../platform';
+import { User as DbUser } from '../models';
+import { resinApi, root, sbvrUtils } from '../platform';
 import { captureException, handleHttpErrors } from '../platform/errors';
 import { resetCounter } from '../lib/rate-limiting';
 import { RequestHandler } from 'express';
 import { SetupOptions } from '..';
 
 const { BadRequestError, NotFoundError } = sbvrUtils;
 
-export const whoami: RequestHandler = (req, res) =>
-	getUser(req)
-		.then(user =>
-			res.send({ id: user.id, username: user.username, email: user.email }),
-		)
-		.catch(err => {
-			if (handleHttpErrors(req, res, err)) {
-				return;
+export const whoami: RequestHandler = async (req, res) => {
+	try {
+		let user: AnyObject = await getUser(req);
```

Comment:
Would it work if we also change this to:
```suggestion
		let user: User & { email?: string } = await getUser(req);
```
Where User is `import { User } from '../platform/jwt';`
PS: we define `type Overwrite<T, U> = Pick<T, Exclude<keyof T, keyof U>> & U;` in common.d.ts and use it here.